src/server/Monte.AuthServer/Services/UserService.cs

```diff
@@ -0,0 +1,188 @@
+﻿using Microsoft.AspNetCore.Identity;
+using Microsoft.AspNetCore.Mvc;
+using Monte.AuthServer.Features.Users.Extensions;
+using Monte.AuthServer.Features.Users.Models;
+using Monte.AuthServer.Helpers;
+
+namespace Monte.AuthServer.Services;
+
+public class UserService : IUserService
+{
+    private readonly UserManager<IdentityUser> _userManager;
+
+    public UserService(UserManager<IdentityUser> userManager)
+    {
+        _userManager = userManager;
+    }
+
+    public async Task<IActionResult> UserCreation(CreateUserRequest request, string role)
```

Comment:
the service should return regular models, which are then wrapped in `IActionResult` in controllers

Reply:
also `CreateUser` would be a more consistent name